plugin-server/src/utils/db/db.ts

```diff
@@ -673,109 +654,70 @@ export class DB {
-            index += 2
-            args.push(groupIdentifier.index, groupIdentifier.key)
-        }
-        const queryResult = await this.postgresQuery<{
-            group_type_index: number
-            group_key: string
-            group_properties: Properties
-            created_at: string
-        }>(
-            'SELECT group_type_index, group_key, group_properties, created_at FROM posthog_group WHERE team_id=$1 AND ('.concat(
-                queryOptions.join(' OR ')
-            ) + ')',
-            args,
-            'getGroupProperties'
-        )
+    public async getGroupsColumns(teamId: number, groupIds: GroupId[]): Promise<Record<string, any>> {
+        const groupColumns: Record<string, any> = {}
 
-        const result: GroupCacheData[] = groupIdentifiers.map((identifier) => {
-            const row = queryResult.rows.filter(
-                (row) => row.group_key === identifier.key && row.group_type_index == identifier.index
-            )[0]
+        for (const [groupTypeIndex, groupKey] of groupIds) {
+            const groupCacheKey = this.getGroupDataCacheKey(teamId, groupTypeIndex, groupKey)
+            const propertiesColumnName = `group${groupTypeIndex}_properties`
+            const createdAtColumnName = `group${groupTypeIndex}_created_at`
 
-            return {
-                identifier,
-                data: row
-                    ? {
-                          properties: row.group_properties,
-                          created_at: DateTime.fromISO(row.created_at).toUTC(),
-                      }
-                    : null,
+            // Lookup data from the cache, but don't throw errors - we'll fallback to Postgres if Redis is unavailable
+            try {
+                const cachedGroupData = await this.redisGet<CachedGroupData | null>(groupCacheKey, null)
+
+                if (cachedGroupData) {
+                    this.statsd?.increment('group_info_cache.hit')
+                    groupColumns[propertiesColumnName] = JSON.stringify(cachedGroupData.properties)
+                    groupColumns[createdAtColumnName] = cachedGroupData.created_at
+                    continue
+                }
+            } catch (error) {
+                captureException(error)
             }
-        })
 
-        // We update the cache for every row, including missing ones to avoid hitting the DB consistently
-        this.promiseManager.trackPromise(this.updateGroupDataCache(teamId, result))
-        return result
-    }
+            this.statsd?.increment('group_info_cache.miss')
 
-    public async fetchGroupColumnsValues(teamId: number, groups: GroupIdentifier[]): Promise<Record<string, string>> {
-        if (groups.length === 0) {
-            return {}
-        }
-        const cachedResults = await Promise.all(
-            groups.map((groupIdentifier) => this.getGroupDataCache(teamId, groupIdentifier))
-        )
+            // If we didn't find cached data, lookup the group from Postgres
```

Comment:
nit: instead of commenting consider having separate functions for getting data from cache and lookup from Postgres, if the function is short and calls clearly named functions the comments aren't needed

Reply:
I'm generally a proponent of this but in this case this function is small enough and "integrated" enough (setting/updating state) that I think this is better than having to jump all around the place to understand what's going on